src/wmtk/Scheduler.cpp

```diff
@@ -13,30 +16,40 @@ Scheduler::~Scheduler() = default;
 SchedulerStats Scheduler::run_operation_on_all(operations::Operation& op)
 {
     SchedulerStats res;
+    std::vector<Simplex> simplices;
 
     const auto type = op.primitive_type();
+    {
+        POLYSOLVE_SCOPED_STOPWATCH("Collecting primitives", res.collecting_time, logger());
+
+        const auto tups = op.mesh().get_all(type);
+        simplices = wmtk::simplex::utils::tuple_vector_to_homogeneous_simplex_vector(tups, type);
+    }
 
-    const auto tups = op.mesh().get_all(type);
 
-    std::vector<Simplex> simplices =
-        wmtk::simplex::utils::tuple_vector_to_homogeneous_simplex_vector(tups, type);
+    logger().info("Executing on {} simplices", simplices.size());
 
-    spdlog::info("Executing on {} simplices", simplices.size());
+    {
+        POLYSOLVE_SCOPED_STOPWATCH("Sorting", res.sorting_time, logger());
```

Comment:
why did this need a macro? if this is just some RAII thing i always just create a class ;-)

Reply:
it comes from polysolve, i will not refactor it.